nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -793,13 +798,21 @@ private void validateState(final FlowFile flowFile) {
             throw new FlowFileHandlingException(flowFile + " is not the most recent version of this flow file within this session");
         }
 
+
         for (final List<MockFlowFile> flowFiles : transferMap.values()) {
             if (flowFiles.contains(flowFile)) {
                 throw new IllegalStateException(flowFile + " has already been transferred");
             }
         }
     }
 
+    private void validateDestinationRelation(final Relationship relationship) {
+        if (!relationships.contains(relationship)) {
+            rollback();
```

Comment:
I think , @pvillard31  's direction was same too. So, I shall make necessary changes. Can you please check what does Standard Process Session does now a day in such situation?